core/src/main/scala/kafka/cluster/Partition.scala

```diff
@@ -499,7 +500,16 @@ class Partition(val topicPartition: TopicPartition,
         addingReplicas = addingReplicas,
         removingReplicas = removingReplicas
       )
-      createLogIfNotExists(partitionState.isNew, isFutureReplica = false, highWatermarkCheckpoints)
+      try {
+        this.createLogIfNotExists(partitionState.isNew, isFutureReplica = false, highWatermarkCheckpoints)
+      } catch {
+        case e: ZooKeeperClientException =>
+          stateChangeLogger.info(s"Because a ZooKeeper client exception has occurred, completed become leader " +
+            s"state change from epoch $leaderEpoch only for those updated partitions with before " +
+            s"ZooKeeper disconnect occurred.", e)
+          error(s"ZooKeeper client occurred while rendering a $topicPartition's leader through zkClient.'", e)
```

Comment:
Do we need ' before "?

Also, the text could probably be sth like "ZooKeeper client error occurred while becoming leader for $topicPartition." 

Ditto in makeFollower().

Reply:
👍 , have made that change. Thanks @junrao